bluemira/equilibria/coils/_grouping.py

```diff
@@ -1015,3 +1157,287 @@ def from_group_vecs(
+
+        Notes
+        -----
+            For FULL and NONE symmetry status, analytic derivatives can be used.
+            When the status is FULL, the derivative values must be halved
+            after applying the repetition matrix as they will be added together.
+            For PARTIAL symmetry status, numerical derivatives must be used.
+        """
+        if all(isinstance(c, SymmetricCircuit) for c in self._coils):
+            return CoilSetSymmetryStatus.FULL
+        if any(isinstance(c, SymmetricCircuit) for c in self._coils):
+            return CoilSetSymmetryStatus.PARTIAL
+        return CoilSetSymmetryStatus.NONE
+
+    @property
+    def _opt_currents_expand_mat(self) -> np.ndarray:
+        """
+        Get the optimisation currents expansion matrix.
+
+        This matrix is used to convert the optimisable currents to the full set of
+        currents in the CoilSet.
+        """
+        cc = self.get_control_coils()
+
+        n_all_coils = cc.n_coils()
+        n_opt_coils = cc.n_current_optimisable_coils
+        n_distinct_coils_and_groupings = len(cc._coils)
+
+        if cc._opt_currents_symmetry_status == CoilSetSymmetryStatus.NONE:
+            return np.eye(n_all_coils)
+
+        # this should be true as, at the top level, the number
+        # of coil or group objects should be the same as the no
+        # of optimisable coils
+        assert n_opt_coils == n_distinct_coils_and_groupings  # noqa: S101
+
+        # you are putting 1's in the col. corresponding
+        # to all coils in the same Circuit
+        mat = np.zeros((n_all_coils, n_opt_coils))
+        i_row = 0
+        for i_col, c in enumerate(cc._coils):
+            if isinstance(c, Circuit):
+                n_coils_in_group = c.n_coils()
+                for n in range(n_coils_in_group):
+                    mat[i_row + n, i_col] = 1
+                i_row += n
+            else:
+                mat[i_row, i_col] = 1
+            i_row += 1
+        return mat
+
+    @property
+    def _opt_currents_sym_reduce_mat(self) -> np.ndarray:
+        """
+        Get the optimisation currents symmetry reduce matrix.
+
+        This matrix is used to convert a full set of optimisation currents
+        into a reduced set be filtering out all non-primary symmetric circuit
+        coil currents.
+        """
+        cc = self.get_control_coils()
+
+        n_all_coils = cc.n_coils()
+        n_opt_coils = cc.n_current_optimisable_coils
+
+        if cc._opt_currents_symmetry_status != CoilSetSymmetryStatus.FULL:
+            raise ValueError(
+                "Symmetry reduce matrix can only be used with a CoilSet "
+                "that only has SymmetricCircuits"
+            )
+
+        mat = np.zeros((n_all_coils, n_opt_coils))
+        i_row = 0
+        for i_col in range(n_opt_coils):
+            mat[i_row, i_col] = 1
+            mat[i_row + 1, i_col] = 0
+            i_row += 2
```

Comment:
I haven't looked at this file in detail, but I notice just that in the corresponding function above, you use `n_coils_in_group`, whereas here you use 2. Is there some mechanism that limits the symmetrycircuits to 2 coils? Should these two related functions be "similar" in implementation, or is there a reason for this difference?

Reply:
Yeah that's well observed.

I can make it clearer with a comment but it's becuase I raise if `cc._opt_currents_symmetry_status != CoilSetSymmetryStatus.FULL:` which means every element in the coilset is a SymmCct. Which means I just need a matrix of alternationg 1's and 0's per col.